openpgp/s2k/s2k.go

```diff
@@ -295,6 +296,14 @@ func (params *Params) Dummy() bool {
 	return params != nil && params.mode == GnuS2K
 }
 
+func (params *Params) salt() []byte {
+	switch params.mode {
+		case SaltedS2K, IteratedSaltedS2K: return params.saltBytes[:8]
+		case Argon2S2K: return params.saltBytes[:Argon2SaltSize]
+		default: return params.saltBytes[:0]
```

Comment:
```suggestion
		default: return nil
```
(or did this not work for some reason?)